### ISSUE TITLE
Should `no-conditional-expect` ignore value-like `expect.*(...)` calls?

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#926]
## Question

Should `no-conditional-expect` ignore value-like static `expect.*(...)` calls such as `expect.any(...)`?

For example:

```ts
it('example', () => {
  expect(value).toEqual({
    message: condition ? expect.any(String) : 'fallback',
  })
})
```

Currently, `vitest/no-conditional-expect` reports `expect.any(String)` with `Avoid calling expect inside conditional statements`, because it is inside a conditional expression.

In this case, however, the actual assertion is the outer `expect(value).toEqual(...)`.
`expect.any(String)` only creates a value used by that assertion, so the assertion itself is always executed.

So I wonder if `no-conditional-expect` should only report actual assertion calls, and not static asymmetric matcher factory calls such as:

```ts
expect.any(String)
expect.anything()
expect.objectContaining({ id: 1 })
expect.arrayContaining([1])
expect.stringContaining('value')
expect.stringMatching(/value/)
expect.closeTo(1, 2)
expect.schemaMatching(schema)
```

This may also need a parser-side distinction. Currently, these value-like static calls can be represented similarly to assertion matcher calls, even though they do not assert anything by themselves.

Would it make sense to distinguish them from actual assertions so rules like `no-conditional-expect` can target only assertion calls?
